pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -108,6 +108,31 @@ type streamIngestionResumer struct {
 	job *jobs.Job
 }
 
+func connectToActiveClient(
+	ctx context.Context, ingestionJob *jobs.Job,
+) (streamclient.Client, error) {
+	details := ingestionJob.Details().(jobspb.StreamIngestionDetails)
+	progress := ingestionJob.Progress()
+	topology := progress.GetStreamIngest().Topology
+
+	if len(topology.PartitionInfo) > 0 {
+		log.Infof(ctx, "ingestion job %d has existing topology, attempting to connect to topology addresses", ingestionJob.ID())
+		client, err := streamclient.NewClientFromTopology(ctx, &topology)
```

Comment:
does the job coordinator always want to connect to the same node in the topology? Or do we need to randomize?

Reply:
Right now we'd always connect to the same one assuming Dial succeeded.  Do we gain anything from randomization?  The serial nature at least makes tests easier.

Reply:
Not sure; I was asking this question... not sure; I don't think it's needed now.